cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -346,8 +343,13 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
             }
         } catch (ForceableProblem e) {
             if (restartContext.backOff.done() || e.forceNow) {
-                log.warn("{}: Pod {} will be force-rolled", reconciliation, podName(podId));
-                restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS);
+                if (canRoll(allClient, podId, 60_000, TimeUnit.MILLISECONDS)) {
```

Comment:
This does not make sense to me. You should not respond to `ForceableProblem` thrown from `canRoll` by running `canRoll` again. Maybe `canRoll` should not throw `ForceableProblem` then?